contracts/protocol/libraries/logic/LiquidationLogic.sol

```diff
@@ -451,13 +451,15 @@ library LiquidationLogic {
             0,
             vars.actualCollateralToLiquidate
         );
-
+        // no time lock needed here
+        DataTypes.TimeLockParams memory timeLockParams;
         // Burn the equivalent amount of xToken, sending the underlying to the liquidator
         IPToken(vars.collateralXToken).burn(
             params.borrower,
             params.liquidator,
             vars.actualCollateralToLiquidate,
-            collateralReserveCache.nextLiquidityIndex
+            collateralReserveCache.nextLiquidityIndex,
+            timeLockParams
```

Comment:
we are injecting the logic into each burn + transferUnderlying except liquidation. it's a lot of changes and hard to be consistent, can we make them all into PToken?